### ISSUE TITLE
queries/terraform/aws/config_rule_for_encrypted_volumes_is_disabled trigger false positive

### ISSUE THREAD

**Issue by jycamier** [Checkmarx/kics#5037]
It seam that `queries/terraform/aws/config_rule_for_encrypted_volumes_is_disabled` is triggering false positive and I don't understand the query.

**Comment by rafaela-soares** [comment]
Hi, @jycamier!

Thank you so much for interacting so much with us! 

Can you give us more details about the possible FP's, please?

Regarding the query itself, it verifies if there is no AWS Config rule for encrypted volumes. As mentioned in the [documentation](https://docs.aws.amazon.com/config/latest/developerguide/encrypted-volumes.html), the identifier `ENCRYPTED_VOLUMES` "checks if the EBS volumes that are in an attached state are encrypted". This rule can help to check if the EBS volumes are not encrypted.

KICS checks if there is an `aws_config_config_rule` resource that does not define a source identifier for `ENCRYPTED_VOLUMES`. 

Let me know if you have further questions 😊

**Comment by jycamier** [comment]
Hi @rafaela-soares 

Thank you for your explanation : now it's very clear in my mind and I think there's a trouble in the query.
I'll prepare a PR to expose you what IMHO it currently doesn't work well.

**Comment by rafaela-soares** [comment]
Hello, @jycamier!

Thank you so much! 🚀

**Comment by jycamier** [comment]
@rafaela-soares 

I'm coming back.
Sadly, I think I can't test what I want to expose due to the way the tests are organized.

So, here's what I think.

## Context

Here's the query

```rego
package Cx

CxPolicy[result] {
	resource := input.document[i].resource
	config := resource.aws_config_config_rule

	not checkSource(config, "ENCRYPTED_VOLUMES")

	result := {
		"documentId": input.document[i].id,
		"searchKey": "aws_config_config_rule", #refer to the first rule
		"issueType": "MissingAttribute",
		"keyExpectedValue": "There is a 'aws_config_config_rule' resource has source id: 'ENCRYPTED_VOLUMES'",
		"keyActualValue": "No 'aws_config_config_rule' resource has source id: 'ENCRYPTED_VOLUMES'",
	}
}

checkSource(config_rules, expected_source) {
	source := config_rules[_].source
	source.source_identifier == expected_source
} else = false {
	true
}
```

This query `checkSource` for each document. However, terraform doesn't work by document but work by layer (and a layer is a directory). It results that this query isn't able to check properly a global layer.

For exemple, let's get a simple layer.

```
00-config-rule
├── my-config-rule-for-x.tf
├── my-config-rule-for-y.tf
├── input.tf
├── output.tf
├── variables.tf
└── providers.tf
```

If you decide to organise your terraform code with your `aws_config_config_rule` which deal with the `ENCRYPTED_VOLUMES` in the file `my-config-rule-for-x.tf` and all other `aws_config_config_rule` in the file `my-config-rule-for-y.tf`, the query trigger an error which fingerpoint the document `my-config-rule-for-y.tf` because of your code organization.

So the solution to test there's one et only one ressource `aws_config_config_rule` which deals with the `ENCRYPTED_VOLUMES`  in the full layer looks-like that : 

```rego
package Cx

CxPolicy[result] {
	input.document[i].resource.aws_config_config_rule

	not checkSource(input, "ENCRYPTED_VOLUMES")

	result := {
		"documentId": input.document[i].id,
		"searchKey": "aws_config_config_rule", #refer to the first rule
		"issueType": "MissingAttribute",
		"keyExpectedValue": "There is a 'aws_config_config_rule' resource has source id: 'ENCRYPTED_VOLUMES'",
		"keyActualValue": "No 'aws_config_config_rule' resource has source id: 'ENCRYPTED_VOLUMES'",
	}
}

checkSource(in, expected_source) {
	source := in.document[_].resource.aws_config_config_rule[_].source
	source.source_identifier == expected_source
} else = false {
	true
}
```

However, it not possible due to a limitation : the test organization.

## Limitation

Today, a test directory is organized like that : 

```
test
├── negative.tf
├── positive.tf
└── positive_expected_result.json
```

To have something which works like terraform work, we need to have a test organization by layer for terraform

```
test
├── negative1
│   └── negative.tf
├── positive1
│   └── positive.tf
└── positive_expected_result.json
```

**Comment by rafaela-soares** [comment]
You are absolutely right, @jycamier! 

Thank you so much for identifying the bug in the query. Such a great observation!

Regarding the tests, it is really a limitation. I will approach that limitation with the whole KICS team as soon as we meet.

**Comment by jycamier** [comment]
Excellent !  thank your so much.

**Comment by rafaela-soares** [comment]
@jycamier, sorry for the inactivity on this issue. The matter is not forgotten. We will take care of it as soon as possible.

**Comment by jycamier** [comment]
No problem @rafaela-soares ! So much work to do and not enough time in a day !
